staging/vhost-device-gpu/src/virt_gpu.rs

```diff
@@ -55,6 +55,121 @@ fn sglist_to_rutabaga_iovecs(
     Ok(rutabaga_iovecs)
 }
 
+pub trait VirtioGpuBackend
+where
+    Self: Sized,
+{
+    fn update_scanout_resource(
+        &mut self,
+        gpu_backend: &mut GpuBackend,
+        gpu_scanout: VhostUserGpuScanout,
+        resource_id: u32,
+        scanout_data: Option<VirtioScanoutBlobData>,
+    ) -> VirtioGpuResult;
+    fn resource_unmap_blob(
+        &mut self,
+        resource_id: u32,
+        shm_region: &VirtioShmRegion,
+    ) -> VirtioGpuResult;
+    fn resource_map_blob(
+        &mut self,
+        resource_id: u32,
+        shm_region: &VirtioShmRegion,
+        offset: u64,
+    ) -> VirtioGpuResult;
+    fn resource_create_blob(
+        &mut self,
+        ctx_id: u32,
+        resource_id: u32,
+        resource_create_blob: ResourceCreateBlob,
+        vecs: Vec<(GuestAddress, usize)>,
+        mem: &GuestMemoryMmap,
+    ) -> VirtioGpuResult;
+    fn process_fence(
+        &mut self,
+        ring: VirtioGpuRing,
+        fence_id: u64,
+        desc_index: u16,
+        len: u32,
+    ) -> bool;
+    fn create_fence(&mut self, rutabaga_fence: RutabagaFence) -> VirtioGpuResult;
+    fn submit_command(
+        &mut self,
+        ctx_id: u32,
+        commands: &mut [u8],
+        fence_ids: &[u64],
+    ) -> VirtioGpuResult;
+    fn context_detach_resource(&mut self, ctx_id: u32, resource_id: u32) -> VirtioGpuResult;
+    fn context_attach_resource(&mut self, ctx_id: u32, resource_id: u32) -> VirtioGpuResult;
+    fn destroy_context(&mut self, ctx_id: u32) -> VirtioGpuResult;
+    fn result_from_query(&mut self, resource_id: u32) -> GpuResponse;
```

Comment:
`result_from_query` doesn't seem to be part of the API of the `VirtioGpu`, like the original comment suggested `  // Non-public function -- no doc comment needed!` suggests.  We can have this function in the implementation of the `VirtioGpu` (in the normal `impl` block without trait).